packages/core/src/tweening/spring.ts

```diff
@@ -0,0 +1,116 @@
+import {decorate, threadable} from '../decorators';
+import {ThreadGenerator} from '../threading';
+import {useProject, useThread, useLogger} from '../utils';
+
+decorate(spring, threadable());
+export function* spring(
+  spring: Spring | null,
+  from: number,
+  to: number,
+  settleTolerance: number | null,
```

Comment:
Thought separating function and spring might be useful – but that's totally made up, so … idk if this is really a good idea.

Reply:
Do you mean the settleTolerance, if so:
The settleTolerance is related to the from & to attributes. So imo it may be better to keep them separate. Some would like to spring between 0 and 1 and so they may want a very small tolerance.

Reply:
Maybe the value of `settleTolerance` should be relative to the distance between `from` and `to`?
It just seems like a rarely used option, so having to write `null` all the time could be a bit annoying.
Alternatively, maybe we could provide multiple signatures for `spring`?
One with `spring, from, to, onProgress` and one with `spring, from, to, settleTolerance, onProgress`

Reply:
Yea, I see, indeed. 
Ok lets try that.